framework_agreement/model/portfolio.py

```diff
@@ -35,15 +42,212 @@ def get_from_supplier(self, supplier):
             return self.create({'supplier_id': supplier.id,
                                 'name': supplier.name})
 
-    name = fields.Char('Name', required=True)
+    def default_name(self):
+        return self.env['ir.sequence'].next_by_code('framework.agreement')
+
+    name = fields.Char('Name', required=True, default=default_name)
     supplier_id = fields.Many2one('res.partner', 'Supplier', required=True)
+    line_ids = fields.One2many('agreement.product.line', 'portfolio_id')
     company_id = fields.Many2one('res.company', 'Company',
                                  default=_company_get)
-    agreement_ids = fields.One2many('framework.agreement', 'portfolio_id',
-                                    'Agreements')
+    pricelist_ids = fields.One2many('product.pricelist', 'portfolio_id',
+                                    'Pricelists')
+    origin = fields.Char('Origin')
+    start_date = fields.Date(required=True, default=fields.Date.today)
+    end_date = fields.Date(help="Leave empty for unlimited duration")
+    draft = fields.Boolean('Is draft')
+    state = fields.Selection(
+        selection=[('draft', 'Draft'),
+                   ('future', 'Future'),
+                   ('running', 'Running'),
+                   ('consumed', 'Consumed'),
+                   ('closed', 'Closed')],
+        string='State',
+        compute='_compute_state',
+        search='_search_state',
+        # not stored because it depends on the current date
+    )
+    purchase_ids = fields.One2many('purchase.order', 'portfolio_id')
 
     _sql_constraints = [
         ('uniq_portfolio',
          'unique(supplier_id, company_id)',
```

Comment:
Really what if a portfolio is expired will not be able to create a new one. You should add a more complex check that use the state